pkg/translator/sumoicmarshaler/sumo_ic_marshaler_test.go

```diff
@@ -0,0 +1,102 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+func TestMarshalerMissingAttributes(t *testing.T) {
+	logs := plog.NewLogs()
+	rl := logs.ResourceLogs().AppendEmpty()
+	rl.ScopeLogs().AppendEmpty()
+	marshaler := &SumoICMarshaler{"txt"}
+	require.NotNil(t, marshaler)
+	_, err := marshaler.MarshalLogs(logs)
+	assert.Error(t, err)
+}
+
+func TestMarshalerMissingSourceHost(t *testing.T) {
+	logs := plog.NewLogs()
+	rls := logs.ResourceLogs().AppendEmpty()
+	rls.Resource().Attributes().InsertString("_sourceCategory", "testcategory")
+
+	marshaler := &SumoICMarshaler{"txt"}
+	require.NotNil(t, marshaler)
+	_, err := marshaler.MarshalLogs(logs)
+	assert.Error(t, err)
+}
+
+func TestMarshalerMissingScopedLogs(t *testing.T) {
+	logs := plog.NewLogs()
+	rls := logs.ResourceLogs().AppendEmpty()
+	rls.Resource().Attributes().InsertString("_sourceCategory", "testcategory")
+	rls.Resource().Attributes().InsertString("_sourceHost", "testHost")
+
+	marshaler := &SumoICMarshaler{"txt"}
+	require.NotNil(t, marshaler)
+	_, err := marshaler.MarshalLogs(logs)
+	assert.NoError(t, err)
+}
+
+func TestMarshalerMissingSourceName(t *testing.T) {
+	logs := plog.NewLogs()
+	rls := logs.ResourceLogs().AppendEmpty()
+	rls.Resource().Attributes().InsertString("_sourceCategory", "testcategory")
+	rls.Resource().Attributes().InsertString("_sourceHost", "testHost")
+
+	sl := rls.ScopeLogs().AppendEmpty()
+	const recordNum = 0
+
+	ts := pcommon.Timestamp(int64(recordNum) * time.Millisecond.Nanoseconds())
+	logRecord := sl.LogRecords().AppendEmpty()
+	logRecord.Body().SetStringVal("entry1")
+	logRecord.SetTimestamp(ts)
+
+	marshaler := &SumoICMarshaler{"txt"}
+	require.NotNil(t, marshaler)
+	_, err := marshaler.MarshalLogs(logs)
+	assert.Error(t, err)
+}
+
+func TestMarshalerOkStructure(t *testing.T) {
+	logs := plog.NewLogs()
+	rls := logs.ResourceLogs().AppendEmpty()
+	rls.Resource().Attributes().InsertString("_sourceCategory", "testcategory")
+	rls.Resource().Attributes().InsertString("_sourceHost", "testHost")
+
+	sl := rls.ScopeLogs().AppendEmpty()
+	const recordNum = 0
+
+	ts := pcommon.Timestamp(int64(recordNum) * time.Millisecond.Nanoseconds())
+	logRecord := sl.LogRecords().AppendEmpty()
+	logRecord.Body().SetStringVal("entry1")
+	logRecord.Attributes().InsertString("log.file.path_resolved", "testSourceName")
+	logRecord.SetTimestamp(ts)
```

Comment:
I would go with map containing all possible types. Can be done in this or separate test

Reply:
ok